### DISCUSSION TITLE
VS code c++ formating isuue

### DISCUSSION THREAD

**Discussion by anirudh7065** [microsoft/vscode-cpptools#11028] [Q&A] [answered]
ms-vscode.cpptools-1.15.4-win32-x64/bin/../LLVM/bin/clang-format.exe -style=clangdiag.clangformatexecutable -fallback-style=LLVM -sort-includes=1 --Wno-error=unknown -assume-filename=E:\.vscode\n.cpp

showing this in the output box what to do

**Comment by sean-mcmanus** [comment]
It sounds like you're hitting a formatting failure. Your formatting settings seem to be incorrect. In particular, -style=clangdiag.clangformatexecutable seems to be invalid. Can you fix your C_Cpp.clang_format_style setting?

**Reply by anirudh7065** [reply]
Can you tell me how can i do that

**Reply by anirudh7065** [reply]
Its showing extension is the source

**Reply by sean-mcmanus** [reply]
Check this setting and clear it:
![image](https://github.com/microsoft/vscode-cpptools/assets/19859882/01a428ac-9882-46d5-b5e9-4bbcc2f4f794)

**Reply by anirudh7065** [reply]
![image](https://github.com/microsoft/vscode-cpptools/assets/107138932/396138a0-94a9-44bd-bc2e-2c3c77f6cbb8)
![image](https://github.com/microsoft/vscode-cpptools/assets/107138932/89eb8b40-1d42-498e-a6b6-a99c6feed408)
![image](https://github.com/microsoft/vscode-cpptools/assets/107138932/39e79aee-bb14-417a-b0ad-2f9be118595b)

**Reply by anirudh7065** [reply]
no results

**Reply by anirudh7065** [reply]
> Check this setting and clear it: ![image](https://user-images.githubusercontent.com/19859882/242982048-01a428ac-9882-46d5-b5e9-4bbcc2f4f794.png)

i cleared it but nothing happened

**Reply by sean-mcmanus** [reply]
@anirudh7065 Actually, due to bug https://github.com/microsoft/vscode-cpptools/issues/10994 you want to make sure you delete the entire setting in the json editor and don't just set it to the empty string (until that bug is fixed in 1.16.1). After doing that does it work? Otherwise, what is the error logging?

**Reply by anirudh7065** [reply]
Actually i dont know how to do that i m beginner right now so i guess i will wait for the update

**Reply by sean-mcmanus** [reply, marked as answer]
Actually, you can use Reset Setting: 
![image](https://github.com/microsoft/vscode-cpptools/assets/19859882/671bb823-7224-4fa5-a009-6fe09d3ea939)

**Reply by anirudh7065** [reply]
> Actually, you can use Reset Setting: ![image](https://user-images.githubusercontent.com/19859882/242990833-671bb823-7224-4fa5-a009-6fe09d3ea939.png)

Actually i have reset every formatting yesterday but I didn't work

**Reply by sean-mcmanus** [reply]
So what is the error you're getting? FYI, there are user/remote/workspace/workspace folder settings, so you may have to check a different scope.

**Reply by anirudh7065** [reply]
> So what is the error you're getting? FYI, there are user/remote/workspace/workspace folder settings, so you may have to check a different scope.

I reset every settings again and it worked finally than you
